torch/_inductor/codegen/cpp_wrapper_cpu.py

```diff
@@ -1422,17 +1422,28 @@ def can_stack_allocate_buffer(self, buffer):
         )
 
     def make_buffer_free(self, buffer):
-        return (
-            ""
-            if isinstance(buffer.get_layout(), ir.MultiOutputLayout)
+        if (
+            isinstance(buffer.get_layout(), ir.MultiOutputLayout)
             or (V.graph.aot_mode and buffer.get_name() in self.stack_allocated_buffers)
             or (
                 config.use_minimal_arrayref_interface
                 and V.graph.aot_mode
                 and buffer.get_name() in V.graph.graph_inputs
             )
-            else f"{buffer.get_name()}.reset();"
-        )
+        ):
+            return ""
+        elif (
+            isinstance(buffer, ir.ExternKernel)
+            and isinstance(buffer.op_overload, torch._ops.OpOverload)
+            and isinstance(buffer.op_overload._schema.returns[0].type, torch.ListType)
+            and isinstance(
+                buffer.op_overload._schema.returns[0].type.getElementType(),
+                torch.TensorType,
+            )
+        ):
+            return f"for (auto t : {buffer.get_name()}) t.reset();"
```

Comment:
You can use reference (`auto& t`) here. Also it's clearer to wrap `t.reset();` with `{}`.